### ISSUE TITLE
upgrade to pmd 6.18.0

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#7225]
details: https://github.com/checkstyle/checkstyle/pull/7214

Huge amount of false positives violations (You have 588 PMD violations) from 
https://pmd.github.io/latest/pmd_rules_java_codestyle.html#unnecessaryfullyqualifiedname
Rule should be disabled.

There are even more false positives:
```
[INFO] --- maven-pmd-plugin:3.12.0:check (default-cli) @ checkstyle ---
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.PropertyCacheFile:220 Rule:UnusedPrivateMethod Priority:3 Avoid unused private methods such as 'getHashCodeBasedOnObjectContent(Serializable)'..
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.filters.CsvFilterElement:71 Rule:UnusedPrivateMethod Priority:3 Avoid unused private methods such as 'addFilter(IntFilterElement)'..
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheckTest:150 Rule:UnusedPrivateMethod Priority:3 Avoid unused private methods such as 'verifyWarns(Configuration,String,String...)'..
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.filters.SuppressWarningsFilterTest:128 Rule:UnusedPrivateMethod Priority:3 Avoid unused private methods such as 'verifySuppressed(Configuration,String...)'..
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.filters.SuppressWithNearbyCommentFilterTest:313 Rule:UnusedPrivateMethod Priority:3 Avoid unused private methods such as 'verifySuppressed(Configuration,String...)'..
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.filters.SuppressWithPlainTextCommentFilterTest:726 Rule:UnusedPrivateMethod Priority:3 Avoid unused private methods such as 'verifySuppressed(String,String,Configuration...)'..
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.filters.SuppressionCommentFilterTest:266 Rule:UnusedPrivateMethod Priority:3 Avoid unused private methods such as 'verifySuppressed(Configuration,String...)'..
```
Rule https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedprivatemethod should be excluded, too much false positives.

----

warnings that appeared after upgrade will be addressed in #7229

**Comment by romani** [comment]
fix is merged.
